instrument_registry/lifespan.py

```diff
@@ -0,0 +1,40 @@
+import json
+import logging
+from collections.abc import AsyncGenerator
+from contextlib import asynccontextmanager
+from typing import Any, TextIO
+
+import yaml
+from fastapi import FastAPI
+from instrument_registry.core.config import PROJECT_NAME, VERSION, get_settings
+from instrument_registry.core.log import configure_logging
+
+cached_data: dict[str, Any] = {}
```

Comment:
i would propably move this to its own singleton class. this allows you a bit of control on how the data is accessed.